bin/s3stalker_daemon.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2022 Pytroll developers
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""S3stalker daemon/runner.
+
+This is a daemon supposed to stay up and running "forever". It will continously
+fetch fresh filenames from an S3 object store and publish the urls of those new
+filenames. It is a daemon version of the s3stalker.py script which needs to be
+run as a cronjob.
+
+"""
+
+import argparse
+import logging
+import logging.config
+from datetime import datetime, timedelta, timezone
+import time
+import yaml
+import sys
+
+from pytroll_collectors.helper_functions import read_yaml
+from pytroll_collectors.s3stalker import create_messages_for_recent_files
+from pytroll_collectors.s3stalker import get_last_fetch
+
+import signal
+from threading import Thread
+from posttroll.publisher import create_publisher_from_dict_config
+
+logger = logging.getLogger(__name__)
+
+
+class S3StalkerRunner(Thread):
+    """Runner for stalking for new files in an S3 object store."""
+
+    def __init__(self, bucket, config, startup_timedelta_seconds):
+        """Initialize the S3Stalker runner class."""
+        super().__init__()
+
+        self.bucket = bucket
+        self.config = config
+        self.startup_timedelta_seconds = startup_timedelta_seconds
+        self.time_back = self.config['timedelta']
+        self.publisher = None
+        self.loop = False
+        self._setup_and_start_communication()
```

Comment:
Would it make sense to automatically start the stalker here? Or move the publisher creation in `.run()`? Now `.close()` call is needed even if the stalker hasn't been started, and that might not be so evident.

Reply:
Yes, you are right! Didn't see that. I have moved the start of the publisher in under the `run` method now.